drastic-price-change-template/src/agent.js

```diff
@@ -0,0 +1,117 @@
+const {
+  Finding,
+  FindingSeverity,
+  FindingType,
+} = require('forta-agent');
+const ARIMA = require('arima');
+
+const { priceDiscrepancyThreshold, asset } = require('../bot-config.json');
+const {
+  getChainlinkPrice,
+  getCoingeckoPrice,
+  getTopTokenHolders,
+  getChainlinkContract,
+  calculatePercentage,
+} = require('./helper');
+
+const INTERVAL = 3600; // 1 hour
+const timeSeries = [];
+
+let lastTimestamp = 0;
+let chainlinkContract;
+let assetDecimals;
+
+// Parameters chosen by finding the model with lowest error and fastest time
+// using the hourly uniswap price from coingecko
+const arima = new ARIMA({
+  p: 5,
+  d: 1,
+  q: 5,
+  verbose: false,
+});
+
+function provideInitialize(getChainlinkContractFn) {
+  return async function initialize() {
+    if (!priceDiscrepancyThreshold
+        || !asset
+        || !asset.chainlinkFeedAddress
+        || !asset.coingeckoId
+        || !asset.contract) {
+      throw new Error('You need to provide valid config file');
+    }
+
+    // Get asset decimals
+    chainlinkContract = getChainlinkContractFn(asset.chainlinkFeedAddress);
+    assetDecimals = await chainlinkContract.decimals();
+  };
+}
+
+function provideHandleBlock(
+  getChainlinkPriceFn,
+  getCoingeckoPriceFn,
+  getTopTokenHoldersFn,
+) {
+  return async function handleBlock(blockEvent) {
+    const findings = [];
+    const { timestamp } = blockEvent.block;
+
+    // Check the price once every INTERVAL
+    if (timestamp < lastTimestamp + INTERVAL) return findings;
+
+    // Get the price from chainlink and coingecko
+    const [chainlinkPrice, coingeckoPrice] = await Promise.all([
+      getChainlinkPriceFn(chainlinkContract, assetDecimals),
+      getCoingeckoPriceFn(asset.coingeckoId),
+    ]);
+
+    if (calculatePercentage(chainlinkPrice, coingeckoPrice) >= priceDiscrepancyThreshold) {
```

Comment:
how would we handle simple lags in price information during volatile times?

Reply:
The chainlink price feeds can be updated when enough time has passed or if a "Deviation threshold" is met. For most of the assets I checked it is <2% so the chainlink price should be update on time